### ISSUE TITLE
How do I just tokenize, without using color themes?

### ISSUE THREAD

**Issue by jameshfisher** [shikijs/shiki#385]
Hi - shiki looks great! I'm trying to use it to replace prism.js on https://tigyog.app/.

But I'm a bit confused about how to just get the raw tokens, or to get HTML output that uses normal CSS variables. I need this so I can generate HTML that uses the existing CSS classes and theme.

I can see `codeToHtml`, but this adds raw CSS colors to the HTML depending on theme. And I can see `codeToThemedTokens`, but that also seems to give me colors, rather than syntactic token types like `variable`, `comment` etc. I eventually found the `'css-variables'` theme, but this seems hacky -- it gives me strings like `"var(--shiki-token-comment)"`, where I just want the string `"comment"`. I could parse the `"comment"` string out of this, but it feels like I'm using it wrongly.

Here's the API I expected:

```ts
interface IToken {
    /**
     * The content of the token
     */
    content: string;
    /**
     * E.g. "comment", "variable", etc
     */
    type: string;

    explanation?: ITokenExplanation[];
}

codeToTokens(code: string, lang?: StringLiteralUnion<Lang>): IToken[][];
```

Basically, I just want to use the tokenizer functionality, without the theme features. Is there a way to do this idiomatically with the shiki API?

**Comment by Gerrit0** [comment]
If you don't want any styling info, you probably don't want Shiki... using vscode-textmate directly will give you the tokens you're after, without attaching theming info. You'll want grammar.tokenizeLine.

You could also set includeExplanation when calling codeToThemedTokens and check the explanation property, but performance will be worse, since that option essentially results in tokenizing everything twice, and slicing strings many times more than without that option

**Comment by jameshfisher** [comment]
@Gerrit0 thanks for the quick reply - I'll look at using vscode-textmate instead.

That said, it seems like Shiki provides other valuable features -- a centralized set of language grammars, and a simpler API. (At a glance, it seems like vscode-textmate leaves you on your own to find the grammars you need, and its API is designed for line-by-line tokenization.) It seems a shame to make those Shiki features unavailable to people unless they also opt into the theming feature.

**Comment by muenzpraeger** [comment]
The big value prop of this project is IMHO that it utilizes VS Code theming.

If you disagree with how it works, then why not fork and modify it to your needs? Or you simply create your own theme (with classnames instead of color codes), and then use that.

**Comment by wooorm** [comment]
Maybe my project https://github.com/wooorm/starry-night might be of help in this case. It’s similar but different. Exposes a AST. Uses the same dependencies.

Also wanted to bring this to the attention to maintainers here: some recent issues are solved by starry-night, so if features don’t really make sense in Shiki, you can send users to starry-night.

**Comment by jameshfisher** [comment]
@wooorm starry-night does exactly what I wanted - I'm now using it in production :-)

(I just had one issue, which I've worked around https://github.com/wooorm/starry-night/issues/9)
